### PR TITLE
Update EVM block timestamp params

### PR DESCRIPTION
**Pull request by saibatizoku**
Uses the `BlockDate` and ledger settings such as `slots_per_epoch`, and `slot_duration`, as well as the `Block0Date` value to calculate the appropriate timestamp each time `apply_block` is called.

### CODE REVIEW

File: chain-impl-mockchain/src/evm/mod.rs [hunk 1]
```diff
@@ -16,7 +16,36 @@ use crate::{
 };
 
 #[cfg(feature = "evm")]
-pub use chain_evm::machine::{BlockGasLimit, Config, Environment, GasPrice};
+pub use chain_evm::machine::{
+    BlockGasLimit, BlockTimestamp, ChainId, Config, Environment, GasPrice,
+};
+
+/// Static parameters for EVM
```
Comment (Zeegomo) [inline chain-impl-mockchain/src/evm/mod.rs:23]: this is static but contains `BlockEpoch` which is a dynamic parameter, have I missed something?

Comment (saibatizoku) [reply chain-impl-mockchain/src/evm/mod.rs:23]: Just a typo, thanks for pointing it out.

**Review by Zeegomo — APPROVED** [review]

File: chain-impl-mockchain/src/ledger/ledger.rs [hunk 2]
```diff
@@ -815,10 +824,10 @@ impl Ledger {
         #[cfg(feature = "evm")]
         {
             // Set EVM environment values derived from block0 values
-            new_ledger.evm.environment.chain_id =
-                <[u8; 32]>::from(new_ledger.static_params.block0_initial_hash).into();
-            // TODO: set Origin, coinbase, and set block timestamp when
-            // they are defined. Using default values meanwhile.
+            new_ledger.evm.environment.chain_id = new_ledger.settings.evm_block_params.chain_id;
+            new_ledger.evm.environment.block_timestamp =
```
Comment (Mr-Leshiy) [inline chain-impl-mockchain/src/ledger/ledger.rs:828]: I think it is better to move it inside `update_evm_block()` function

Comment (saibatizoku) [reply chain-impl-mockchain/src/ledger/ledger.rs]: This has been completely re-worked to go inside a method.

**Review by Mr-Leshiy — APPROVED** [review]

**Comment by minikin** [comment]
@saibatizoku  @Mr-Leshiy  
Can we resolve issues here to merge it, please? Or do we have some other dependencies ?

**Pull request merged**